docs/task_manager.rst

```diff
@@ -0,0 +1,97 @@
+Background Task Processing
+######################################
+
+.. currentmodule:: task_manager
+
+Introduction
+======================================
+
+In order to build responsive Toolkit apps, work needs to be scheduled to run
+in background threads. The :class:`BackgroundTaskManager` handles generic
+background processing and is used by many of the Toolkit frameworks.
+
+When for example using a :class:`~shotgun_model.ShotgunModel`, the model
+will internally create a :class:`~shotgun_data.ShotgunDataRetriever` to handle
+background communication with Shotgun. The :class:`~shotgun_data.ShotgunDataRetriever`
+in turn uses threads to handle its
+background processing and uses this :class:`BackgroundTaskManager` in order
+to schedule the background processing.
+
+A centralized thread pool
+-------------------------
+
+By default, each Shotgun Model will have its own background task manager
+that runs Shotgun queries in the background. If your app uses a lot of
+different models or data retrievers, this becomes hard to maintain and may
+lead to unpredictable states due to the fact that the Shotgun API isn't
+thread safe.
+
+.. note::
+
+    We recommend that apps using several Shotgun Models or Data retrievers
+    use a single  :class:`BackgroundTaskManager` for handling of all its
+    background processing.
+
+In these situations, you can maintain a single :class:`BackgroundTaskManager`
+for your app and pass this manager in upon instantiation of your Shotgun
+Models and data retriever. This allows for a setup where all threaded work
```

Comment:
Maybe this instead: "for your app and supply it to your :class:`~shotgun_model.ShotgunModel` and :class:`~shotgun_data.ShotgunDataRetriever` instances when creating them."